jopa-impl/src/main/java/cz/cvut/kbss/jopa/sessions/ServerSession.java

```diff
@@ -94,6 +95,12 @@ private void initialize(OntologyStorageProperties storageProperties, Configurati
      * @return UnitOfWork instance
      */
     public UnitOfWork acquireUnitOfWork(Configuration configuration) {
+        final String enabledStr = configuration.get(JOPAPersistenceProperties.TRANSACTION_MODE);
+        if (enabledStr != null && enabledStr.equals("read_only")) {
```

Comment:
If you reverse the equals (`"read_only".equals(enabledStr)`), you don't need the nullcheck.

Reply:
Also, I would prefer the `read_only` to be defined in `Constants` together with explanatory Javadoc instead of just literal used here and mentioned in `JOPAPersistenceProperties` docs.